cmd/authd/daemon/migration_test.go

```diff
@@ -0,0 +1,109 @@
+package daemon
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMigrateOldCacheDir(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		oldDirExists     bool
+		newDirExists     bool
+		oldDirUnreadable bool
+		newDirUnreadable bool
+
+		wantErr          bool
+		wantOldDirExists bool
+		wantNewDirExists bool
+	}{
+		"Success if old dir does not exist": {oldDirExists: false, newDirExists: false},
+		"Success if old dir exists and new dir does not": {
+			oldDirExists:     true,
+			newDirExists:     false,
+			wantOldDirExists: false,
+			wantNewDirExists: true,
+		},
+		"Success if old dir exists and new dir exists": {
+			oldDirExists:     true,
+			newDirExists:     true,
+			wantOldDirExists: true,
+			wantNewDirExists: true,
+		},
+		"Success if old dir exists but is unreadable": {
+			oldDirExists:     true,
+			oldDirUnreadable: true,
+		},
+
+		"Error if new dir exists but is unreadable": {
+			oldDirExists:     true,
+			newDirExists:     true,
+			newDirUnreadable: true,
+			wantErr:          true,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			oldParentDir := t.TempDir()
+			newParentDir := t.TempDir()
+			oldDir := filepath.Join(oldParentDir, "cache")
+			newDir := filepath.Join(newParentDir, "cache")
+
+			if tc.oldDirExists {
+				err := os.Mkdir(oldDir, 0700)
+				require.NoError(t, err)
+
+				if tc.oldDirUnreadable {
+					err := os.Chmod(oldParentDir, 0000)
+					require.NoError(t, err)
+					// Ensure that the directory is readable after the test, to avoid issues with cleanup
+					defer func() {
+						//nolint:gosec // G302 Permissions 0700 are not insecure for a directory
+						err := os.Chmod(oldParentDir, 0700)
+						require.NoError(t, err)
```

Comment:
Here and in the others `require.NoError()` calls maybe add a `Setup:`, `TearDown:` (or any other) context to explain it better if they happen

Reply:
My opinion is still that the source file + line printed by `require.NoError()` on error already allows us to quickly go to this line in an IDE, which provides better context than any message we could add - except if the message contains the value of some variable which might help debugging. If you disagree, I'll add a message, but I really don't see any value in it.

Reply:
done as requested [below](https://github.com/ubuntu/authd/pull/676#discussion_r1871236643)